File: example2/ping.py
import optparse
import phish
import time

def message_callback(message):
  global count
  count += 1
  if count < options.count:
    phish.send(ping)
  else:
    phish.close_port(0)

def closed_callback():
  print "Elapsed time for %s pingpong of %s bytes = %s secs" % (count, len(ping), time.clock() - start_time)
  phish.loop_complete()

phish.init()
phish.enable_input_port(0, message_callback, closed_callback)
phish.enable_output_port(0)
phish.check

parser = optparse.OptionParser()
parser.add_option("--count", type="int", default=1000, help="Number of pings.  Default: %default.")
parser.add_option("--size", type="int", default=0, help="Number of bytes in each ping.  Default: %default.")
(options, arguments) = parser.parse_args()

count = 0
ping = "*" * options.size
start_time = time.clock()

phish.send(ping)
phish.loop()

File: minnow2/ping.cpp
#include <phish.hpp>

int count = 0;
int max_count = 100000;
std::string ping = "";

void message_callback(uint32_t part_count)
{
  count += 1;
  if(count < max_count)
  {
    phish::pack(ping);
    phish::send();
  }
  else
  {
    phish::close_port(0);
  }
}

int main(int argc, char* argv[])
{
  phish::init(argc, argv);
  phish::input(0, message_callback, phish::loop_complete);
  phish::output(0);
  phish::check();

  phish::timer::wallclock timer;

  phish::pack(ping);
  phish::send();
  phish::loop();

  std::cerr << "Elapsed time for " << count << " pingpong of " << ping.size() << " bytes = " << timer.elapsed() << " secs" << std::endl;

  phish::close();
}
